### PR TITLE
CM-29: Added download for beneficiaries

### PR DESCRIPTION
**Pull request by dborowiecki**
Ticket: [CM-29](https://openimis.atlassian.net/browse/CM-29)

Requires https://github.com/openimis/openimis-be-social_protection_py/pull/5

[CM-29]: https://openimis.atlassian.net/browse/CM-29?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
-    fetchedBeneficaries,
+    fetchedBeneficiaries,
     errorBeneficiaries,
     beneficiaries,
     beneficiariesPageInfo,
     beneficiariesTotalCount,
     status,
-    readOnly
+    readOnly,
+
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:28]: Please remove this empty line space

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 2]
```diff
@@ -64,41 +73,91 @@ const BenefitPlanBeneficiariesSearcher = ({
...
 
+    const [failedExport, setFailedExport] = useState(false)
+
+    useEffect(() => {
+        setFailedExport(true)
+      }, [errorBeneficiaryExport])
+
+    useEffect(() => {
+    if (beneficiaryExport) {
+        downloadExport(beneficiaryExport, 'beneficiary_export.csv')();
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:84]: I think 'beneficiary_export.csv' file name could be handled by constant.

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 3]
```diff
@@ -64,41 +73,91 @@ const BenefitPlanBeneficiariesSearcher = ({
...
                 sorts={sorts}
                 rowsPerPageOptions={ROWS_PER_PAGE_OPTIONS}
                 defaultPageSize={DEFAULT_PAGE_SIZE}
                 defaultFilters={defaultFilters()}
+                cacheFiltersKey={"benefitPlanBeneficiaryFilterCache"}
             />
+            {failedExport && (
+                <Dialog fullWidth maxWidth="sm">
+                  <DialogTitle>{errorBeneficiaryExport}</DialogTitle>
+                  <DialogActions>
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:132]:
```

                   <DialogActions>
                     <Button onClick={setFailedExport(false)} variant="contained">
                       {formatMessage(intl, "socialProtection", "ok")}
                     </Button>
                   </DialogActions>

```
Please fix indentation

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
-import {formatMessageWithValues, Searcher} from "@openimis/fe-core";
+import {formatMessage, formatMessageWithValues, Searcher, downloadExport} from "@openimis/fe-core";
 import {bindActionCreators} from "redux";
 import {connect} from "react-redux";
-import {fetchBeneficiaries} from "../actions";
-import {DEFAULT_PAGE_SIZE, ROWS_PER_PAGE_OPTIONS} from "../constants";
+import {fetchBeneficiaries, downloadBeneficiaries} from "../actions";
+import {DEFAULT_PAGE_SIZE, ROWS_PER_PAGE_OPTIONS, RIGHT_BENEFICIARY_SEARCH} from "../constants";
 import BenefitPlanBeneficiariesFilter from "./BenefitPlanBeneficiariesFilter";
+import {Button,
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:9]:
```
import {
  Button,
  Dialog,
  DialogActions,
  DialogTitle,
} from "@material-ui/core";
```
You can fix the way of declaring imports to satisfy some standards defined by eslint rules regarding imports.

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 5]
```diff
@@ -64,41 +73,91 @@ const BenefitPlanBeneficiariesSearcher = ({
...
+                exportFetch={downloadBeneficiaries}
+                exportFields={[
+                    'id',
+                    'individual.first_name',
+                    'individual.last_name',
+                    'individual.dob',
+                    'individual.date_created',
+                    'json_ext' // Unfolded by backend and removed from csv
+                ]}
+                exportFieldsColumns={{
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:113]:
```
                exportFieldsColumns={{
                    'id': 'ID',
                    'individual__first_name': formatMessage(intl, "socialProtection", "export.firstName"),
                    'individual__last_name': formatMessage(intl, "socialProtection", "export.lastName"),
                    'individual__dob': formatMessage(intl, "socialProtection", "export.dob"),
                    'individual__date_created': formatMessage(intl, "socialProtection", "export.dateCreated"),
                }}
```
Please align such curly brackets.

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 6]
```diff
@@ -64,41 +73,91 @@ const BenefitPlanBeneficiariesSearcher = ({
         return filters;
     };
 
+    const [failedExport, setFailedExport] = useState(false)
+
+    useEffect(() => {
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:78]:
```
     useEffect(() => {
         setFailedExport(true);
     }, [errorBeneficiaryExport])
```
Indentations and lack of semicolon

File: src/components/BenefitPlanBeneficiariesSearcher.js [hunk 7]
```diff
@@ -64,41 +73,91 @@ const BenefitPlanBeneficiariesSearcher = ({
         return filters;
     };
 
+    const [failedExport, setFailedExport] = useState(false)
```
Comment (sniedzielski) [inline src/components/BenefitPlanBeneficiariesSearcher.js:76]: missing semicolon at the end of line

**Review by sniedzielski — COMMENTED** [review]
LGTM, there is nothing blocking for me at this point. I just added some stylistic comments

**Comment by dborowiecki** [comment]
@sniedzielski All issues are regarding style error. In a long term it'll be not efficient to discover them during the review. I'll attach eslint to the project. As I'll remove all style issues in the process I think we can merge this PR.

**Comment by sniedzielski** [comment]
I definitely aggree, it is a good moment to attach eslint at this stage of project.

**Review by sniedzielski — APPROVED** [review]

**Pull request merged**